modules/dynamodb/src/main/scala/dynamodb/model.scala

```diff
@@ -98,9 +102,96 @@ object model {
 
   case class UpdateExpression(value: String)
 
+  // TODO Macro to instantiate it from static string
+  case class ExpressionAlias private (value: String)
+
+  object ExpressionAlias {
+
+    class Macros(val c: whitebox.Context) {
+      import c.universe._
+
+      def literal(s: c.Expr[String]): Tree =
+        s.tree match {
+          case Literal(Constant(s: String)) =>
+            ExpressionAlias
+              .fromString(s)
+              .fold(
+                e => c.abort(c.enclosingPosition, e),
+                _ =>
+                  q"_root_.com.ovoenergy.comms.aws.dynamodb.model.ExpressionAlias.unsafeFromString($s)"
+              )
+          case _ =>
+            c.abort(
+              c.enclosingPosition,
+              s"This method uses a macro to verify that a String literal is a valid ExpressionAlias. Use ExpressionAlias.fromString if you have a dynamic String that you want to parse."
+            )
+        }
+    }
+
+    def apply(s: String): ExpressionAlias = macro ExpressionAlias.Macros.literal
+
+    def fromString(str: String): Either[String, ExpressionAlias] =
+      if (str.headOption.contains('#')) {
+        new ExpressionAlias(str).asRight
+      } else {
+        s"Valid expression alias must start with '#'. Invalid placeholder: '$str'".asLeft
+      }
+
+    def unsafeFromString(str: String): ExpressionAlias =
+      fromString(str).getOrElse(
+        throw new IllegalArgumentException(
+          s"$str is not a valid expression alias")
+      )
+  }
+
+  // TODO Macro to instantiate it from static string
+  case class ExpressionPlaceholder private (value: String)
+
+  object ExpressionPlaceholder {
+
+    class Macros(val c: whitebox.Context) {
```

Comment:
can these two methods be unified?

Reply:
Definitely, I have noticed there is quite a bit of repetition.